src/renderer/components/modals/CreateMultisigAccount/components/SelectSignatories.tsx

```diff
@@ -0,0 +1,244 @@
+
+    if (tab === SignatoryTabs.WALLETS) setSelectedWallets(newSelectedValues);
+    if (tab === SignatoryTabs.CONTACTS) setSelectedContacts(newSelectedValues);
+
+    onSelect(newWallets, newContacts);
+  };
+
+  const isSameItemSelected = (index: string, selectionMap?: { [index: string]: boolean }): boolean => {
+    if (!selectionMap) return false;
+
+    return Object.entries(selectionMap).some(([key, value]) => {
+      return key === index ? false : value;
+    });
+  };
+
+  const searchedContactList = contactList.filter((c) => {
+    return includes(c.address, query) || includes(c.matrixId, query) || includes(c.name, query);
+  });
+
+  const hasWallets = Boolean(walletList.length);
+  const hasContacts = Boolean(contactList.length);
+
+  const selectedWalletsLength = Object.values(selectedWallets).reduce((acc, w) => {
+    return acc + Object.values(w).reduce((total, flag) => total + (flag ? 1 : 0), 0);
+  }, 0);
+  const selectedContactsLength = Object.values(selectedContacts).reduce((acc, c) => {
+    return acc + Object.values(c).reduce((total, flag) => total + (flag ? 1 : 0), 0);
+  }, 0);
+
+  const WalletsTab = hasWallets ? (
+    <ul className="gap-y-2">
+      {walletList.map(({ index, accountId, name, walletName, chainId }) => (
+        <li key={index + 'wallets'} className="p-1 mb-0.5 last:mb-0 rounded-md hover:bg-action-background-hover">
+          <Checkbox
+            checked={selectedWallets[accountId]?.[index] || false}
+            disabled={isSameItemSelected(index, selectedWallets[accountId])}
+            onChange={() => selectSignatory(SignatoryTabs.WALLETS, index, accountId, selectedWallets)}
+          >
+            <WalletsTabItem
+              name={name}
+              accountId={accountId}
+              walletName={walletName}
+              explorers={chainId ? chains[chainId]?.explorers : []}
+            />
+          </Checkbox>
+        </li>
+      ))}
+    </ul>
+  ) : (
+    <EmptyContacts description={t('createMultisigAccount.noWalletsLabel')} />
+  );
+
+  const ContactsTab = (
+    <div>
+      <div className="flex items-center gap-x-4 flex-1 mb-4">
+        <SearchInput
+          placeholder={t('createMultisigAccount.searchContactPlaceholder')}
+          wrapperClass={cn('flex-1')}
+          value={query}
+          onChange={setQuery}
+        />
+        {hasContacts && (
+          <Button variant="text" suffixElement={<Icon name="add" size={16} />} onClick={toggleContactModalOpen}>
+            {t('createMultisigAccount.addContact')}
+          </Button>
+        )}
+      </div>
+
+      {hasContacts ? (
+        <ul className="flex flex-col gap-y-2">
+          {searchedContactList.map(({ index, accountId, name }) => (
+            <li key={index + 'contacts'} className="p-1 mb-0.5 last:mb-0 rounded-md hover:bg-action-background-hover">
+              <Checkbox
+                checked={selectedContacts[accountId]?.[index] || false}
```

Comment:
Removed `useForm` because there is actually no form 😊
Explicitly added map with selected values